Enhance navbar

RanitDERIA opened the pull request:
## 📋 Description

This pull request enhances the navbar in the CampX web application to provide a better user experience and improved accessibility.

## 🔨 Changes Made

- Implemented responsive design enhancements to ensure the navbar adapts seamlessly across all screen sizes.
- Added hover effects and improved the link arrangement for better clarity and usability.
- Introduced dropdown menus for better organization of content.

## ✅ Checklist

Before submitting the PR, please make sure you have completed the following:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] My changes generate no new warnings.
- [x] I have tested it locally and it works fine.

## 🏷️ Types of Changes

What type of changes does your code introduce? (Check all that apply)

- [x] UI enhancement (non-breaking change which enhances UI) 🎨

## 📸 Screenshots
*Large Screen:*
![Screenshot (409)](https://github.com/user-attachments/assets/6ddb42d7-bbbe-40d5-8f4f-034b637657f5)

*Medium Screen:*
![bfs](https://github.com/user-attachments/assets/2141b7e6-9b8e-49e4-afc5-2bd16e8d615a)

*Small Screen:*
![Screenshot 2024-10-08 002222](https://github.com/user-attachments/assets/3d626888-d052-497d-864d-9a89da5f6412)


*Thank you for contributing to VigneshDevHub! We look forward to reviewing your PR and getting it merged!*

Vignesh025 commented:
The navbar looks good before scrolling, but please change the color of the navbar after scrolling to match the current navbar color. Make any necessary adjustments accordingly.

RanitDERIA commented:
Thank you for the feedback!  I’ll update the navbar color after scrolling to match the current color scheme and make the necessary adjustments accordingly. I'll ensure it maintains consistency and improves the overall design.

I’ll get it done shortly!

RanitDERIA commented:
I have update the navbar color after scrolling to match the current color scheme and make the necessary adjustments accordingly. Please check and suggest any required changes specifically.

Vignesh025 commented:
Hey, did you just replace the navbar on the campgrounds page with the one from the homepage?

RanitDERIA commented:
No, I didn't replace the navbar from the homepage. I made some adjustments to the campgrounds page navbar to improve its functionality and design while keeping it consistent with the overall style of the site.

Vignesh025 commented:
Hey @RanitDERIA ,

Could you please make the following changes to the navbar?

Reduce the height of the navbar.
Change the navbar color to black.
Modify the dropdown styles to remove the underlines, as they don't look good.
Create a separate stylesheet called navbar.css instead of having inline styles.
Thanks!

RanitDERIA commented:
Thanks for the feedback! I've made the requested changes:
- Reduced the navbar height.
- Changed the navbar color to black.
- Updated the dropdown styles to remove underlines.
- Moved all styles into a separate `navbar.css` file.

Everything has been resolved now. Please review and let me know if anything else is needed!

Vignesh025 commented:
Hey @RanitDERIA,

please make the following changes:

- The "Camp" from the CampX logo on the navbar disappears when hovered over, fix it.
- On mobile screens, the dropdown button in the navbar is not visible , make sure it's visible.
- There are quite a few unnecessary styles; please remove all of those.

RanitDERIA commented:
Hey @Vignesh025 ,

Thanks for the feedback! I've made the following updates:

- Fixed the issue where the "Camp" part of the CampX logo disappeared on hover.
- Ensured the dropdown button is now visible on mobile screens.
- Removed unnecessary styles for a cleaner and more efficient design.

I hope this meets all your requirements. Please review and let me know if any further adjustments are needed!

Vignesh025 commented on public/stylesheets/navbar.css:
```diff
@@ -0,0 +1,108 @@
+/* General Navbar Styles */
+.navbar {
+    background-color: black; /* Black background for the navbar */
+    transition: background-color 0.3s ease-in-out;
```
what is the use of this transition?

RanitDERIA replied:
Smooth transition for background color when it changes, enhancing user experience

Vignesh025 replied:
When exactly is the background color supposed to change? It currently stays black all the time.

RanitDERIA replied:
Actually i wrote this when initially i changed the color to white to get a transition. Now its not required so i removed it. Thanks for the feedback !

Vignesh025 commented on public/stylesheets/navbar.css:
```diff
@@ -0,0 +1,108 @@
+    color: rgba(255, 255, 255, 0.5); /* Semi-transparent text */
+    font-weight: 700;
+    margin-left: 1rem;
+    transition: color 0.3s ease;
```
what is the use of this transition?

RanitDERIA replied:
The transition: color 0.3s ease; was applied to ensure that any change in the text color (such as on hover or interaction) would happen smoothly over 0.3 seconds, rather than instantly. This subtle animation improves the visual feedback for users, making the interface feel more responsive and polished. However, since it's removed now, the color will change instantly without the smooth transition.

Vignesh025 commented on public/stylesheets/navbar.css:
```diff
@@ -0,0 +1,108 @@
+}
+
+.navbar-nav .nav-link.active {
+    color: white !important; /* Active link stays white */
```
even without !important , its still white. Then why do you have !important?

RanitDERIA replied:
Removed now

Vignesh025 reviewed (commented):
sync your fork

RanitDERIA commented:
Hey @Vignesh025 ,

I've synced my fork with the upstream and resolved all conflicts. Everything should now be up-to-date and ready for review!

Vignesh025 commented:
Hey @RanitDERIA , make the PR to the test branch and not the main branch
